src/frequenz/sdk/actor/_run_utils.py

```diff
@@ -19,26 +19,15 @@ async def run(*actors: Actor) -> None:
         actors: the actors to be awaited.
     """
     _logger.info("Starting %s actor(s)...", len(actors))
-    await _wait_tasks(
-        set(asyncio.create_task(a.start(), name=str(a)) for a in actors),
-        "starting",
-        "started",
-    )
 
-    # Wait until all actors are done
-    await _wait_tasks(
-        set(asyncio.create_task(a.wait(), name=str(a)) for a in actors),
-        "running",
-        "finished",
-    )
-
-    _logger.info("All %s actor(s) finished.", len(actors))
+    for actor in actors:
+        if actor.is_running:
+            _logger.info("Actor %s: already running.", actor)
+        else:
+            actor.start()
```

Comment:
Optional, but looking at it now I like the symmetry of having both messages, if it was really started or not.

```suggestion
            _logger.info("Actor %s: Starting...", actor)
            actor.start()
```